### PR TITLE
Feature/fix match updates

### PR DESCRIPTION
**Pull request by sstraatemans**
Fixes #
It fixes that an added match is added to the dashboard right away.
and it sets the Wins / Loses in the user branch in firebase. this way we can easier see, who is in what match

## Proposed Changes
- adding wins / loses matches to the user branch in firebase

Sorry it looks like a lot of changes, but I added prettier to my project in VSCode

### CODE REVIEW

File: src/actions/matches.js [hunk 1]
```diff
@@ -2,48 +2,139 @@ import { database } from './../store';
...
+  const model = matchModel({
+    competitors: users
+  });
+  let updateData = {};
+  updateData[`matches/${matchId}`] = model;
+
+  //also update the match data for the competitors
+  users.forEach(userId => {
+    updateData[`users/${userId}/matches/${matchId}`] = true;
+  });
```
Comment (Randynamic) [inline src/actions/matches.js]: Don't you think something like this is more readable

```
      const updateData = users.reduce((object, userId) => { 
        object[`users/${userId}/matches/${matchId}\`] = true; 
        return object;
      }, {`matches/${matchId}`: model});
```

File: src/actions/matches.js [hunk 2]
```diff
@@ -2,48 +2,139 @@ import { database } from './../store';
...
+};
+
+//the match needs to be removed from /matches
+//and the different users
+export const removeMatch = (matchId, match) => {
+  const ref = database.ref('/');
+  let updateData = {};
+  updateData[`matches/${matchId}`] = null;
+  Object.values(match.competitors).map(competitor => {
```
Comment (Randynamic) [inline src/actions/matches.js]: reduce

File: src/components/Match/index.js [hunk 3]
```diff
@@ -5,99 +5,98 @@ import { declareWinner, removeMatch } from './../../actions/matches';
 import { MatchWrapper } from './styles';
 
 class Match extends Component {
+  constructor(props) {
+    super(props);
+    this.handleDeclareWinner = this.handleDeclareWinner.bind(this);
+    this.handleRemoveMatch = this.handleRemoveMatch.bind(this);
```
Comment (Randynamic) [inline src/components/Match/index.js]: unnecessary constructor
use the arrow functions for binding.

File: src/components/Match/index.js [hunk 4]
```diff
@@ -5,99 +5,98 @@ import { declareWinner, removeMatch } from './../../actions/matches';
...
-            }
-        }
-        return false;
+  /**
+   * check if the auth has voted for the competitor
+   * so we can show that on the component
+   * the votes are in the match.winners property
+   */
+  checkAuthVote(competitor) {
+    if (this.props.match.winners && this.props.match.winners[this.props.auth.uid]) {
```
Comment (Randynamic) [inline src/components/Match/index.js]:
```
if (this.props.match.winners && 
    this.props.match.winners[this.props.auth.uid] && 
    this.props.match.winners[this.props.auth.uid] === competitor.uid) {
```

File: src/actions/matches.js (hunk 2 above)
Comment (sstraatemans) [reply src/actions/matches.js]: you cant reduce an object. updateData is an object.
because firebase does everything with objects

Comment (sstraatemans) [reply src/actions/matches.js]: Nevermind

Comment (Randynamic) [reply src/actions/matches.js]: hehehehe if you can map you can also reduce 🤣

**Pull request merged**